### PR TITLE
Add new `defaultTreeId` prop to specify default tree for TreeWidgetUiItemsProvider

### PR DESCRIPTION
**Pull request by pateljay1397**
- Added the change for setting up defaultTree from the viewer component
- Updated the naming for "model" : "models"

### CODE REVIEW

File: packages/itwin/tree-widget/public/locales/en/TreeWidget.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "label2D": "2D",
   "label3D": "3D",
   "learnMore": "Learn more",
-  "modeltree": "Model Tree",
+  "modelstree": "Models Tree",
```
Comment (aruniverse) [inline packages/itwin/tree-widget/public/locales/en/TreeWidget.json:12]:
```suggestion
  "models": "Models",
```

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx [hunk 2]
```diff
@@ -51,9 +51,9 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
       const trees: SelectableContentDefinition[] = [];
 
       if (!this._treeWidgetOptions?.hideTrees?.modelsTree) {
-        trees.push({
-          label: TreeWidget.translate("modeltree"),
-          id: "model-tree",
+        const modelTree = {
+          label: TreeWidget.translate("modelstree"),
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx:55]:
```suggestion
          label: TreeWidget.translate("models"),
```

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx [hunk 3]
```diff
@@ -51,9 +51,9 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
...
+          id: "models-tree",
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx:56]: lets extract these ids we've hardcoded for each of the trees and put them into a constant that we export in the types.ts file. then we can just use those constants here, and others can use those as well

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx [hunk 4]
```diff
@@ -51,9 +51,9 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
       const trees: SelectableContentDefinition[] = [];
 
       if (!this._treeWidgetOptions?.hideTrees?.modelsTree) {
-        trees.push({
-          label: TreeWidget.translate("modeltree"),
-          id: "model-tree",
+        const modelTree = {
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx:54]: im thinking we  change this back, and just continue to push all the trees like before.

and add at the end, if a defaultTree is specified, we just move that up to the front

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx (hunk 3 above)
Comment (pateljay1397) [reply packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx]: I have created the three constants for all three hardcoded treeIds in type.ts

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx (hunk 2 above)
Comment (pateljay1397) [reply packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx]: Updated the change as per the comment

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx (hunk 4 above)
Comment (pateljay1397) [reply packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx]: Added the change to the  end for setting up defaultTree if user have setup

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx [hunk 5]
```diff
@@ -98,6 +98,12 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
         trees.push(...this._treeWidgetOptions.additionalTrees);
       }
 
+      if(this._treeWidgetOptions?.defaultTree && trees.length !== 0){
+        //Adding the defaultTree to first index
+        const defaultTreeId = this._treeWidgetOptions.defaultTree;
+        trees.unshift(trees.splice(trees.indexOf(trees.filter(tree => tree.id===defaultTreeId)[0]), 1)[0]);
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx:104]: nice; but can we break this out from being one liner for better readability?

Comment (pateljay1397) [reply packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx]: Updated the code format based on comments.

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx [hunk 6]
```diff
@@ -98,6 +98,14 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
         trees.push(...this._treeWidgetOptions.additionalTrees);
       }
 
+      if(this._treeWidgetOptions?.defaultTree && trees.length !== 0){
+        //Adding the defaultTree to first index
+        const defaultTreeId = this._treeWidgetOptions.defaultTree;
+        const extractedDefaultTreeId = trees.filter(tree => tree.id===defaultTreeId)[0];
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx:104]:
```suggestion
        const extractedDefaultTree = trees.filter(tree => tree.id===defaultTreeId)[0];
```

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx [hunk 7]
```diff
@@ -98,6 +98,14 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
...
+        const index = trees.indexOf(extractedDefaultTreeId);
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx:105]:
```suggestion
        const index = trees.indexOf(extractedDefaultTree);
```

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx (hunk 6 above)
Comment (pateljay1397) [reply packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx]: Changed "extractedDefaultTreeId" with "extractedDefaultTree"

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx (hunk 7 above)
Comment (pateljay1397) [reply packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx]: Changed "extractedDefaultTreeId" with "extractedDefaultTree"

**Review by aruniverse — APPROVED** [review]
Missing `rush change`. `minor` bump is fine.

you will be prompted for the changelog file entry, this will be visible to the user so just a heads up.

File: common/changes/@itwin/tree-widget-react/jaykumarpatel-defaultTree-view_2022-09-12-18-17.json [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/tree-widget-react",
+      "comment": "Added the code change for the setting up defaultTree from the viewer component. ",
```
Comment (aruniverse) [inline common/changes/@itwin/tree-widget-react/jaykumarpatel-defaultTree-view_2022-09-12-18-17.json:5]:
```suggestion
      "comment": "Add new `defaultTreeId` prop to specify default tree for TreeWidgetUiItemsProvider",
```

File: common/changes/@itwin/tree-widget-react/jaykumarpatel-defaultTree-view_2022-09-12-18-17.json [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+      "type": "patch"
```
Comment (aruniverse) [inline common/changes/@itwin/tree-widget-react/jaykumarpatel-defaultTree-view_2022-09-12-18-17.json:6]:
```suggestion
      "type": "minor"
```

File: packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx [hunk 10]
```diff
@@ -98,9 +98,9 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
         trees.push(...this._treeWidgetOptions.additionalTrees);
       }
 
-      if(this._treeWidgetOptions?.defaultTree && trees.length !== 0){
+      if(this._treeWidgetOptions?.defaultTreeId && trees.length !== 0){
         //Adding the defaultTree to first index
-        const defaultTreeId = this._treeWidgetOptions.defaultTree;
+        const defaultTreeId = this._treeWidgetOptions.defaultTreeId;
```
Comment (aruniverse) [inline packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx:103]:
```suggestion
        const { defaultTreeId } = this._treeWidgetOptions;
```

**Pull request merged**